### PR TITLE
Better bucket texture (Overlay support)

### PR DESCRIPTION
**Pull request by Rui914**
Update bucket texture, also support overlay texture.
Based on #564

### CODE REVIEW

**Comment by PilzAdam** [comment]
I don't like this very much. I would prefer a simple overlay (`bucket_bucket.png^bucket_water.png`, where bucket_water.png is just the blue part). I think your way also breaks existing mods?

**Comment by Rui914** [comment]
@PilzAdam 
Update.
How about this?

**Comment by asl97** [comment]
@PilzAdam imo, that sound like using overlaying just for the sake of overlaying. also if someone change the overlay texture, it could mess up the texture for other mods.

File: mods/bucket/init.lua [hunk 1]
```diff
@@ -46,6 +46,10 @@ function bucket.register_liquid(source, flowing, itemname, inventory_image, name
 	bucket.liquids[flowing] = bucket.liquids[source]
 
 	if itemname ~= nil then
+		if groups and groups.overlay_texture then
+			inventory_image = "bucket_overlay.png^"..inventory_image.."^[noalpha^bucket_overlay.png^[makealpha:255,126,126"
```
Comment (asl97) [inline mods/bucket/init.lua]: unless i am wrong, the first `bucket_overlay.png` would get fully overlay/cover by the inventory_image so there is no reason for it.

**Comment by paramat** [comment]
-1. Lets go back to the sinple implementation, there's only 3 buckets as default and new bucket textures are as easy to make, possibly easier than working with overlays. Overlays are for eliminating huge numbers of combinations such as ores in different stones.

**Comment by C1ffisme** [comment]
3 buckets as default, but, how many more mods with buckets are there?

I have my own personal animals mod with milk, as do many other mods, and there was that oil mod...

**Pull request closed without merging**